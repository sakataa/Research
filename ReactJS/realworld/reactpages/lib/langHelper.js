const DEFAULT_RESOURCE_NAME = "_language_en_US";
const langCode = window.PageLanguage || "en-US";
const resourceName = `_language_${langCode.replace("-", "_")}`;

export default class LangHelper {
    static get currentLanguage() {
        return langCode;
    }

    static getSingleResource(key) {
        var langHandler = window[resourceName];

        return langHandler[key] || window[DEFAULT_RESOURCE_NAME][key] || key;
    }
}