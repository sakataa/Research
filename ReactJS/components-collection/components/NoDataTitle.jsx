import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import langHelper from '../lib/langHelper';

export default class NoDataTitle extends Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <p className="no-info-row center-align">{langHelper.getSingleResource("NoInfo")}</p>
        )
    }
}