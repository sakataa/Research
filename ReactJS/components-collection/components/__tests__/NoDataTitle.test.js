import React from 'react'
import NoDataTitle from '../NoDataTitle'
import renderer from 'react-test-renderer'
import langHelper from '../../lib/langHelper';

describe('NoDataTitle ', () => {
    afterAll(() => {
        jest.resetAllMocks();
    })

    it('component render correctly', () => {
        langHelper.getSingleResource = jest.fn().mockReturnValueOnce("No Data");

        const component = renderer.create(<NoDataTitle />);

        const result = component.toJSON();

        expect(result).toMatchSnapshot();
    });
})