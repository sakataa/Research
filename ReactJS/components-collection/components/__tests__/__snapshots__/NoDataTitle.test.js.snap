// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoDataTitle  component render correctly 1`] = `
<p
  className="no-info-row center-align"
>
  No Data
</p>
`;
